src/main/logutils.ts

```diff
@@ -117,41 +123,173 @@ const tailFile = (path: string) => {
 }    
 
 /**
+ * Splits a WoW combat line intelligently with respect to quotes,
+ * lists, tuples, and what have we.
+ * 
+ */
+const splitLogLine = (line: string): any => {
```

Comment:
The reason it looks like C code (although it's a port from a Python script I made) is that it does something low level that JS cannot do by itself.

It basically parses the log file format _accurately_ and creates arrays as needed for things like affix lists, converts the tuples from talents and such into lists.

I understand the need to keep things simple, but parsing this log file isn't simple as such. Splitting on comma without regard to encounters that contain commas is a recipe for trouble.

Example is the boss name `Skulloc, Son of Gruul` - this previously cut it short to `Skulloc`, which isn't correct.

> I'm even more nervous if this can handle the COMBATANT_DATA (which are fairly complex lines) events we rely on for getting specs/teams in arena.

This is exactly what it can do. I'd be happy to implement some tests for this, if you want.

Reply:
Ah, good point. Look forward to discussing further. You might be interested in taking a look at this: https://github.com/wowarenalogs/wowarenalogs/tree/main/packages/wow-combat-log-parser/src

Reply:
That looks very overengineered, considering it's a partial, heavily PvP centric/specific parser, at a glance.